REST/src/main/java/pl/jakubowskiprzemyslaw/REST/services/FleetGeneratorService.java

```diff
@@ -0,0 +1,140 @@
+
+        boardSize = shipConfiguration.getBoardSize();
+    }
+
+    public Fleet generateRandomFleet() {
+        emptyFields = new EmptyFields(boardSize);
+
+        Fleet fleet = new Fleet();
+        List<Integer> shipPlacementList = shipConfiguration.getShipPlacementList();
+
+        for (Integer shipLength : shipPlacementList) {
+            Ship ship = placeShip(shipLength);
+            fleet.addShip(ship);
+            removeShipAndBufferFromEmptyFields(ship);
+        }
+        return fleet;
+    }
+
+    private Ship placeShip(Integer shipLength) {
+        Optional<Ship> ship = Optional.empty();
+
+        while (!ship.isPresent()) {
+            Coordinate randomCoordinate = emptyFields.getRandomCoordinate();
+            ship = tryToPlaceShip(shipLength, randomCoordinate);
+        }
+
+        return ship.get();
+    }
+
+    private Optional<Ship> tryToPlaceShip(int shipLength, Coordinate coordinate) {
+
+        List<Coordinate> shipCoordinates;
+        try {
+            shipCoordinates = tryToAllocateCoordinatesToBoard(shipLength, coordinate);
+        } catch (CannotAllocatePositionsOnBoardException cannotAllocatePositionsOnBoardException) {
+            return Optional.empty();
+        }
+
+        Ship ship = createShip(shipCoordinates);
+
+        return Optional.of(ship);
+    }
+
+    private List<Coordinate> tryToAllocateCoordinatesToBoard(int shipLength, Coordinate coordinate) throws CannotAllocatePositionsOnBoardException {
+        List<Coordinate> shipCoordinates = createShipCoordinatesWithFirstCoordinateAdded(coordinate);
+
+        Coordinate direction = getDirection();
+
+        Coordinate nextCoordinate = new Coordinate(coordinate.getX(), coordinate.getY());
+        while (shipLength > 1) {
+            nextCoordinate = nextCoordinate.translate(direction);
+            if (!emptyFields.isCoordinateAvailable(nextCoordinate)) {
+                throw new CannotAllocatePositionsOnBoardException("");
+            }
+            shipLength--;
+            shipCoordinates.add(nextCoordinate);
+        }
+
+        return shipCoordinates;
+    }
+
+    private void removeShipAndBufferFromEmptyFields(Ship ship) {
+        List<Coordinate> shipCoordinates = ship.getCoordinates();
+
+        Set<Coordinate> occupiedFields = createBuffer(shipCoordinates);
+        occupiedFields
+                .forEach(emptyFields::removeCoordinate);
+    }
+
+    private Ship createShip(List<Coordinate> shipCoordinates) {
+        Ship ship = new Ship();
+        shipCoordinates.stream()
+                .map(Mast::new)
+                .forEach(ship::addMastToShip);
+        return ship;
+    }
+
+    private Coordinate getDirection() {
+        Random random = new Random();
+        int coordinateX = random.nextInt(2);
+        return new Coordinate(coordinateX, 1 - coordinateX);
+    }
+
+    private List<Coordinate> createShipCoordinatesWithFirstCoordinateAdded(Coordinate coordinate) {
```

Comment:
I would move this method body higher, that way it could be closer and in good order with its calling.

Reply:
done